From 5fd9826b8cdcb9130b0bf6943a2a5cd7c7499370 Mon Sep 17 00:00:00 2001
From: Elizabeth Figura <zfigura@codeweavers.com>
Date: Wed, 27 Aug 2025 17:51:51 -0500
Subject: [PATCH 2/2] quartz: Preserve the current visibility in
 IVideoWindow::put_Style().

This fixes videos in some VisualArts applications.
Based on a patch by Attila Fidan.
---
 dlls/quartz/tests/videorenderer.c | 14 +++++++-------
 dlls/quartz/tests/vmr7.c          | 14 +++++++-------
 dlls/quartz/tests/vmr9.c          | 14 +++++++-------
 dlls/quartz/window.c              |  8 ++++++++
 4 files changed, 29 insertions(+), 21 deletions(-)

diff --git a/dlls/quartz/window.c b/dlls/quartz/window.c
index 02c6a826bb2..684c05245a6 100644
--- a/dlls/quartz/window.c
+++ b/dlls/quartz/window.c
@@ -234,6 +234,14 @@ HRESULT WINAPI BaseControlWindowImpl_put_WindowStyle(IVideoWindow *iface, LONG s
     if (!window->pPin->peer)
         return VFW_E_NOT_CONNECTED;
 
+    /* Preserve the current visibility.
+     * Using IsWindowVisible() has the side effect that a visible child of an
+     * invisible window will be hidden by this call, but that's what native
+     * does. */
+    style = (style & ~WS_VISIBLE);
+    if (IsWindowVisible(window->hwnd))
+        style |= WS_VISIBLE;
+
     SetWindowLongW(window->hwnd, GWL_STYLE, style);
     SetWindowPos(window->hwnd, 0, 0, 0, 0, 0,
             SWP_NOSIZE | SWP_NOMOVE | SWP_NOZORDER | SWP_NOACTIVATE | SWP_FRAMECHANGED);
-- 
GitLab

